### DISCUSSION TITLE
Is there a way to bypass the verification on baseUrl or get it to use a different path for verification?

### DISCUSSION THREAD

**Discussion by jpierson-at-riis** [cypress-io/cypress#25833] [Questions and Help]
I'm running into a problem with an application which does not serve any response on the `baseUrl` (ex. `http://localhost:8080`). Instead the application is served from `http://localhost:8080/my-app/somepage.html`. Due to the normally helpful checks that Cypress does however to see if the server specified by baseUrl is reachable may tests fail to start and I get the following type of error message.

```
Cypress could not verify that this server is running:

  > http://localhost:8080

We are verifying this server because it has been configured as your baseUrl.

Cypress automatically waits until your server is accessible before running tests.

We will try connecting to it 3 more times...
We will try connecting to it 2 more times...
We will try connecting to it 1 more time...

Cypress failed to verify that your server is running.

Please start this server and then run Cypress again.
```

It is expected behavior for my application that a request to `http://localhost:8080` would return a 404 status. How do I proceed to use Cypress without modifying how my application works?

**Comment by portyu9** [comment]
Hi @jpierson-at-riis 

Rather than bypassing the check, configure `baseUrl` to a URL prefix that your application actually serves.

For example, if this responds successfully:

```text
http://localhost:8080/my-app/
```

use:

```js
const { defineConfig } = require('cypress')

module.exports = defineConfig({
  e2e: {
    baseUrl: 'http://localhost:8080/my-app/',
  },
})
```

and then:

```js
cy.visit('somepage.html')
```

Cypress will resolve that relative URL against the configured `baseUrl`.

If **only** the exact HTML URL responds and there is no usable application prefix that Cypress can use as `baseUrl`, the practical alternative without changing the application is to omit `baseUrl` and visit the fully-qualified URL:

```js
cy.visit('http://localhost:8080/my-app/somepage.html')
```

There is not a separate documented option that says “verify `/health` but use a different `baseUrl` for tests”.

So I would either:

1. make the reachable application prefix the `baseUrl`, or
2. omit `baseUrl` and use the complete URL if no suitable prefix responds.

Please mark this answer as accepted if it helped, thank you.
